nextstrain/cli/command/login.py

```diff
@@ -0,0 +1,127 @@
+
+
+def register_arguments(parser):
+    parser.add_argument(
+        "--check",
+        help    = "Check credentials by logging into nextstrain.org",
+        action  = 'store_true')
+
+    return parser
+
+
+def run(opts):
+    if opts.check:
+        print("Checking credentials by logging into nextstrain.org...")
+        username = config.get('login', 'username', fallback=None, path=config.SECRETS)
+
+        if not username:
+            login()
+            return 0
+
+        tokens = {
+            k : config.get('login', k, fallback=None, path=config.SECRETS)
+                for k in [ 'access_token', 'id_token', 'refresh_token' ]
+        }
+
+        u = Cognito(COGNITO_USER_POOL_ID, COGNITO_CLIENT_ID, username=username,
+                id_token=tokens['id_token'],
+                refresh_token=tokens['refresh_token'],
+                access_token=tokens['access_token'])
+
+        try:
+            u.verify_token(tokens['id_token'], 'id_token', 'id')
+            u.verify_token(tokens['access_token'], 'access_token', 'access')
+
+        except (jose.exceptions.JWSError, jose.exceptions.JWTError):
+            print("Invalid credentials.")
+            print()
+            login()
+            return 0
+
+        access_token_expired = u.check_token(tokens['access_token'])
+
+        if not access_token_expired:
+            print("Logged in succesfully!")
+
+        else:
+            print("Replacing expired access token...")
+            tokens['AccessToken'] = u.access_token
+
+            save_credentials({**{"username": username}, **tokens}, secrets_path)
+
+    else:
+        # First try reading credentials from a local secrets file
+        username = config.get('login', 'username', fallback=None, path=config.SECRETS)
+
+        if username:
+            print("Logged into nextstrain.org as %s" % username)
+            print("Log out with `nextstrain logout`")
+
+        else:
+            login()
+
+
+def login():
+    """
+    With user input, logs a Nextstrain Groups user into nextstrain.org.
+
+    Failed login attempts are printed to stdout and result in a non-zero exit
+    code.
+
+    Saves SRP credentials to a local secrets path.
+    """
+    print("Logging into nextstrain.org...")
+    username = input('Username: ')
+    password = getpass.getpass()
+
+    try:
+        u = Cognito(COGNITO_USER_POOL_ID, COGNITO_CLIENT_ID, username=username)
+        u.authenticate(password=password)
+
+    except Exception as e:
+        raise NextstrainUserError("Failed to login: %s" % e)
+
+    print("Logged in successfully!")
+    tokens = {
+        'id_token'          : u.id_token,
+        'refresh_token'     : u.refresh_token,
+        'access_token'      : u.access_token,
+        # TODO the pycognito docs say the Cognito instance should have the
+        # following "expires_*" attributes, but it doesn't.
+        # 'expires_in'        : u.expires_in,
+        # 'expires_datetime'  : u.expires_datetime,
```

Comment:
I'm not sure what to do here, as it appears this is in direct contradiction with pycognito's docs, but I'll take a closer look again tomorrow.